src/MoBi.Presentation/Presenter/ObjectPathCreatorAtTransport.cs

```diff
@@ -123,13 +128,21 @@ private string getPathKeywordForContainer(IContainer container)
          if (container == null)
             return string.Empty;
 
-         if (Transport.TargetCriteria.IsSatisfiedBy(container))
+         if (ProcessBuilder.TargetCriteria.IsSatisfiedBy(container))
             return ObjectPathKeywords.TARGET;
 
-         if (Transport.SourceCriteria.IsSatisfiedBy(container))
+         if (ProcessBuilder.SourceCriteria.IsSatisfiedBy(container))
             return ObjectPathKeywords.SOURCE;
 
          return string.Empty;
       }
+
+      protected override TransportBuilder GetProcessBuilderFor(IParameter parameter)
+      {
+         if (parameter.IsAtTransport())
+            return parameter.RootContainer as TransportBuilder;
+
+         throw new MoBiException($"cant find transport for parameter {parameter.Name}");
```

Comment:
@rwmcintosh WOndering why we start to put error message directly into the code like this. I have seen in multiple times. Is this our new way of doing exception instead of collating the text in one central place?